### PR TITLE
[FIX] Add function to prep data for JSON serialization

### LINKED ISSUE #858: Object of type ndarray is not JSON serializable
### Summary

This issue is the result of a failing test in #839 as reported by @handwerkerd: https://github.com/ME-ICA/tedana/pull/839#issuecomment-1069523729

It looks like our `save_json` function is having trouble saving a numpy array to the json dictionary.

### Next Steps

As mentioned by @tsalo, we might need to automatically convert numpy arrays to lists in `save_json`.

### PR DESCRIPTION
**Pull request by jbteves**
Closes #858  .

Changes proposed in this pull request:

- adds prep_data_for_json to inspect a dictionary for ndarrays and change them into lists for saving data to JSON
-

### CODE REVIEW

**Review by eurunuela — DISMISSED** [review]
LGTM! Thanks @jbteves

**Review by eurunuela — APPROVED** [review]

File: tedana/io.py
Comment (handwerkerd) [inline tedana/io.py:706]:
```suggestion
    Currently converts numpy arrays to lists and raises an error if there's
    another data structure that can be automatically written to JSON.
    Use this to make something serializable when writing JSON to disk.
```

Really minor suggested change to make clear this is really only for correcting one thing right now (accept or reject this suggestion). Otherwise LGTM.

**Review by handwerkerd — CHANGES_REQUESTED** [review]

File: tedana/io.py
Comment (tsalo) [reply tedana/io.py:706]: I like this suggestion, but I think it should say "another data structure that _cannot_ be automatically written to JSON."

Comment (jbteves) [reply tedana/io.py:706]: I think it's distinctly possible that more could come, and I'd rather not have to change the docstring and lead people to believe they should expand the docstring rather than just chucking another type into the if-block below.

Comment (handwerkerd) [reply tedana/io.py:706]: Like I said. I'm fine other way. The plus is, if I look at in a year, the note will help me remember why this function was needed. The minus is, as you note, it might over-promise future additions or over-specify. Resolve this review with or without making the change and I'll then approve.

Comment (jbteves) [reply tedana/io.py]: Latest commit modifies the language to make it clear that this function should be used to add more conversion cases without specifying them, that feels pretty split down the middle.

**Review by handwerkerd — APPROVED** [review]

**Comment by handwerkerd** [comment]
@eurunuela You previously approved this PR, but there were a few changes and it needs a re-approval. This is what's holding up getting the PCA curves accessible in tedana. If you get a chance, can you check this out and re-approve. If you're in transit, maybe @tsalo can look at this? Thank you.

**Comment by eurunuela** [comment]
Will try to have a look before Friday.

**Review by eurunuela — APPROVED** [review]
LGTM! Thank you @jbteves !

**Pull request merged**
